chimney/src/test/scala/io/scalaland/chimney/IssuesSpec.scala

```diff
@@ -760,4 +760,14 @@ class IssuesSpec extends ChimneySpec {
     import Issue498.*
     (Foo.Sub1("test"): Foo).into[Bar].withFieldConst(_.b, 1).transform ==> Bar.Sub1("test", 1)
   }
+
+  test("support non-sealed trait destination") {
+    (colors1.Green: colors1.Color)
+      .into[String]
+      .withSealedSubtypeHandled[colors1.Red.type](_ => "red")
+      .withSealedSubtypeHandled[colors1.Green.type](_ => "green")
+      .withSealedSubtypeHandled[colors1.Blue.type](_ => "blue")
+      .transform ==> "green"
```

Comment:
Hi again @MateuszKubuszok,

While working on this test, I suddenly realised that this is not much different than

```scala
(colors1.Green: colors1.Color) match {
  case colors1.Red => "red"
  case colors1.Green => "green"
  case colors1.Blue => "blue"
}
```

or if I wanted to use Transformer, for the subtypes I can do that:

```scala
(colors1.Green: colors1.Color) match {
  case red:colors1.Red.type => red.into[Foo].transform
  case green:colors1.Green.type => green.into[Foo].transform
  case blue:colors1.Blue.Type => blue.into[Foo].transform
}
```

so I am happy to drop this PR, since we can achieve the same the same result and with the same effort, using regular pattern matching...

Reply:
yeah, that would work. I will close this PR